straxen/plugins/peak_shadow.py

```diff
@@ -0,0 +1,79 @@
+import numpy as np
+import strax
+import numba
+export, __all__ = strax.exporter()
+
+
+@export
+@strax.takes_config(
+    strax.Option('pre_s2_area_threshold', default=1000,
+                 help='Only take S2s large than this into account'
+                 ' when calculating PeakShadow [PE]'),
+    strax.Option('time_window_backward', default=int(3e9),
+                 help='Search for S2s causing shadow in this time window [ns]'),
+    strax.Option('skip_drift_time', default=0,
+                 help='Deduct max drift time to avoid peak interference in a single event [ns]'))
+class PeakShadow(strax.OverlapWindowPlugin):
+    """
+    This plugin can find and calculate the previous S2 shadow at peak level,
+    with time window backward and previous S2 area as options.
+    It also gives the area and position infomation of these previous S2s.
+    """
+
+    __version__ = '0.0.2'
+    depends_on = ('peak_basics', 'peak_positions')
+    provides = 'peak_shadow'
+    save_when = strax.SaveWhen.EXPLICIT
+
+    def get_window_size(self):
+        return self.config['time_window_backward']
```

Comment:
You can also make the window a bit larger than actually needed. In this way you can avoid some boundary conditions more easily.